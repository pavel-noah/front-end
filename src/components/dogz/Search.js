import React from 'react';
import PropTypes from 'prop-types';

export default function Search({ searchInput, onChange, submitSearch }) {
    return (<>
            <form onSubmit={event => submitSearch(event, searchInput)}>
                Search For A Dog By Name: <input type="text" name="searchInput" value={searchInput} onChange={onChange}></input>
                <button>Search</button>
            </form>
        </>
    );
}

Search.propTypes = {
    searchInput: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    submitSearch: PropTypes.func.isRequired
};
