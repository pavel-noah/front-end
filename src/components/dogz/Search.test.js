import React from 'react';
import { shallow } from 'enzyme';
import Search from './Search';

describe('Search', () => {
    it('renders search', () => {
        const wrapper = shallow(<Search submitSearch={() => {}} onChange={() => {}} searchInput="" />);
        expect(wrapper).toMatchSnapshot();
    });
});
  
