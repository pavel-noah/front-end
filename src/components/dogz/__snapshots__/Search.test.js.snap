// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders search 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    Search For A Dog By Name: 
    <input
      name="searchInput"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button>
      Search
    </button>
  </form>
</Fragment>
`;
